Two packages are providing parts to the `chip`module, and there are imports from one into the other. The packages are

- home-assistant-chip-core
- home-assistant-chip-clusters

In the test phase of `python-matter-server` the `home-assistant-chip-core` package tries to import bits that should be provided by `home-assistant-chip-clusters`, but it fails to find them.

In `python-matter-server` the `home-assistant-chip-clusters` package is propagated, and the `home-assistant-chip-core` package is optional and passed into `checkInputs`.

```
_________________ ERROR collecting tests/server/test_server.py _________________
ImportError while importing test module '/build/source/tests/server/test_server.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/server/test_server.py:8: in <module>
    from chip.clusters import ClusterCommand
/nix/store/92n7kjlhvda17lnvnal47alnq4y5gw8y-python3.10-home-assistant-chip-core-2022.12.0/lib/python3.10/site-packages/chip/clusters/__init__.py:24: in <module>
    from . import Command
/nix/store/92n7kjlhvda17lnvnal47alnq4y5gw8y-python3.10-home-assistant-chip-core-2022.12.0/lib/python3.10/site-packages/chip/clusters/Command.py:26: in <module>
    from .ClusterObjects import ClusterCommand
E   ModuleNotFoundError: No module named 'chip.clusters.ClusterObjects'
=========================== short test summary info ============================
ERROR tests/server/test_server.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
```

When I create a nix-shell using `python3.withPackages (ps: with ps; [ home-assistant-chip-core home-assistant-chip-clusters ])` I can do the import just fine.

```
❯ nix-shell -I nixpkgs=$(pwd) -p "python3.withPackages (ps: with ps; [ home-assistant-chip-clusters home-assistant-chip-core ])"
this derivation will be built:
  /nix/store/j6hk7yk3vhkn7201ansy0ys7834zrd67-python3-3.10.9-env.drv
building '/nix/store/j6hk7yk3vhkn7201ansy0ys7834zrd67-python3-3.10.9-env.drv'...
created 466 symlinks in user environment
❯ python3
Python 3.10.9 (main, Dec  6 2022, 18:44:57) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from chip.clusters import ClusterCommand
>>> 
```

Why does it fail during check phase?

--- a/pkgs/development/python-modules/home-assistant-chip-clusters/default.nix
+++ b/pkgs/development/python-modules/home-assistant-chip-clusters/default.nix
@@ -1,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, dacite
+}:
+
+buildPythonPackage rec {
+  pname = "home-assistant-chip-clusters";
+  version = "2022.12.0";
+  format = "wheel";
+
+  src = fetchPypi {
+    inherit format version;
+    pname = "home_assistant_chip_clusters";
+    dist = "py3";
+    python = "py3";
+    hash = "sha256-zo54lRNyKXCCUUoYIiZmHZMqISim9QKEOnFbM/iBRqE=";
+  };
+
+  propagatedBuildInputs = [
+    dacite
+  ];
+
+  pythonImportsCheck = [
+    "chip.clusters"
+  ];
+
+  doCheck = false; # no tests
+
+  meta = with lib; {
+    description = "Python-base APIs and tools for CHIP";
+    homepage = "https://github.com/home-assistant-libs/chip-wheels";
+    license = licenses.asl20;
+    maintainers = teams.home-assistant.members;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+  };
+}

--- a/pkgs/development/python-modules/home-assistant-chip-core/default.nix
+++ b/pkgs/development/python-modules/home-assistant-chip-core/default.nix
@@ -1,0 +1,93 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+
+# build
+, autoPatchelfHook
+
+# runtime
+, openssl_1_1
+
+# propagates
+, coloredlogs
+, construct
+, dacite
+, rich
+, pyyaml
+, ipdb
+, deprecation
+, mobly
+, pygobject3
+}:
+
+buildPythonPackage rec {
+  pname = "home-assistant-chip-core";
+  version = "2022.12.0";
+  format = "wheel";
+
+  disabled = pythonOlder "3.7";
+
+  src = let
+    system = {
+      "aarch64-linux" = {
+        name = "aarch64";
+        hash = "sha256-oNqrvbzXeXpMG3v9RK6kppONH4n7xLVaJCEFXxVj2jE=";
+      };
+      "x86_64-linux" = {
+        name = "x86_64";
+        hash = "sha256-S5n1MUig8ZDSLgWeVmu+5qLZ4kfHQUC9qZcVfM8rPvw=";
+      };
+    }.${stdenv.system} or (throw "Unsupported system");
+  in fetchPypi {
+    pname = "home_assistant_chip_core";
+    inherit version format;
+    dist = "cp37";
+    python = "cp37";
+    abi = "abi3";
+    platform = "manylinux_2_31_${system.name}";
+    hash = system.hash;
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    openssl_1_1
+  ];
+
+  propagatedBuildInputs = [
+    coloredlogs
+    construct
+    dacite
+    rich
+    pyyaml
+    ipdb
+    deprecation
+    mobly
+    pygobject3
+  ];
+
+  pythonImportsCheck = [
+    "chip"
+    "chip.ble"
+    # https://github.com/project-chip/connectedhomeip/pull/24376
+    #"chip.configuration"
+    "chip.discovery"
+    "chip.native"
+    "chip.storage"
+  ];
+
+  doCheck = false; # no tests
+
+  meta = with lib; {
+    description = "Python-base APIs and tools for CHIP";
+    homepage = "https://github.com/home-assistant-libs/chip-wheels";
+    license = licenses.asl20;
+    maintainers = teams.home-assistant.members;
+    platforms = platforms.linux;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+  };
+}

--- a/pkgs/development/python-modules/mobly/default.nix
+++ b/pkgs/development/python-modules/mobly/default.nix
@@ -1,0 +1,51 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+# runtime
+, portpicker
+, pyserial
+, pyyaml
+, timeout-decorator
+, typing-extensions
+
+# tests
+, procps
+, pytestCheckHook
+, pytz
+}:
+
+buildPythonPackage rec {
+  pname = "mobly";
+  version = "1.12";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "mobly";
+    rev = "refs/tags/${version}";
+    hash = "sha256-HAXm0/h5jbgVuIwP7IZ1ffUs92gcpOPiM2VgT38r8Go=";
+  };
+
+  propagatedBuildInputs = [
+    portpicker
+    pyserial
+    pyyaml
+    timeout-decorator
+    typing-extensions
+  ];
+
+  checkInputs = [
+    procps
+    pytestCheckHook
+    pytz
+  ];
+
+  meta = with lib; {
+    changelog = "https://github.com/google/mobly/blob/";
+    description = "Automation framework for special end-to-end test cases";
+    homepage = "https://github.com/google/mobly";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ hexa ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4297,6 +4297,8 @@ self: super: with self; {
 
   home-assistant-bluetooth = callPackage ../development/python-modules/home-assistant-bluetooth { };
 
+  home-assistant-chip-clusters = callPackage ../development/python-modules/home-assistant-chip-clusters { };
+
   homeassistant-pyozw = callPackage ../development/python-modules/homeassistant-pyozw { };
 
   homeconnect = callPackage ../development/python-modules/homeconnect { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5888,6 +5888,8 @@ self: super: with self; {
 
   moat-ble = callPackage ../development/python-modules/moat-ble { };
 
+  mobly = callPackage ../development/python-modules/mobly { };
+
   mocket = callPackage ../development/python-modules/mocket { };
 
   mock = callPackage ../development/python-modules/mock { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4299,6 +4299,8 @@ self: super: with self; {
 
   home-assistant-chip-clusters = callPackage ../development/python-modules/home-assistant-chip-clusters { };
 
+  home-assistant-chip-core = callPackage ../development/python-modules/home-assistant-chip-core { };
+
   homeassistant-pyozw = callPackage ../development/python-modules/homeassistant-pyozw { };
 
   homeconnect = callPackage ../development/python-modules/homeconnect { };

